src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -194,7 +194,7 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                         }
                         else
                         {
-                            _response.Headers.TryAddWithoutValidation(descriptor, headerValue);
+                            _response.Headers.TryAddWithoutValidation(descriptor.HeaderType == HttpHeaderType.Request ? descriptor.AsCustomHeader() : descriptor, headerValue);
```

Comment:
Do we need to do the same thing for trailing headers?

Reply:
updated code and tests to cover that as well.